datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1316,23 +1353,43 @@ const MAX_LIST_VALUE_SIZE_REWRITE: usize = 20;
 /// expression that will evaluate to FALSE if it can be determined no
 /// rows between the min/max values could pass the predicates.
 ///
+/// Any predicates that can not be translated will be passed to `unhandled_hook`.
+///
 /// Returns the pruning predicate as an [`PhysicalExpr`]
 ///
-/// Notice: Does not handle [`phys_expr::InListExpr`] greater than 20, which will be rewritten to TRUE
+/// Notice: Does not handle [`phys_expr::InListExpr`] greater than 20, which will fall back to calling `unhandled_hook`
```

Comment:
Since this is a new API I think it might be worth thinking a bit more about. What would you think about moving this logic into its own structure? 

For example, instead of

```rust
        let known_expression = col("a").eq(lit(ScalarValue::Int32(Some(12))));
        let known_expression_transformed = rewrite_predicate_to_statistics_predicate(
            &logical2physical(&known_expression, &schema),
            &schema,
            None,
        );
```

Something like

```rust
        let known_expression = col("a").eq(lit(ScalarValue::Int32(Some(12))));
        let known_expression_transformed = StatisticsPredicateRewriter::new()
           .rewrite(
             &logical2physical(&known_expression, &schema)
             &schema)
           );
```


Or with a rewrite hook

```rust
            let expr = logical2physical(&expr, &schema_with_b);
            StatisticsPredicateRewriter::new()
              .with_unhandledhook(|expr| {
                 // closure that handles expr
              })
           .rewrite(
                &expr,
                &schema,
            )
```

Reply:
Yup that sounds good to me, I'll push that change tomorrow.

Reply:
Actually just did it, it was easy.

My main question is if it should include the schema or not.